src/compiler/checker/lines_4000_8000.rs

```diff
@@ -542,6 +572,35 @@ impl NodeBuilder {
         }
     }
 
+    fn map_to_type_nodes(
+        &self,
+        type_checker: &TypeChecker,
+        types: Option<&[Rc<Type>]>,
+        context: &NodeBuilderContext,
+        is_bare_list: Option<bool>,
+    ) -> Option<Vec<Rc<Node /*TypeNode*/>>> {
+        if let Some(types) = types {
+            if !types.is_empty()
+            /*some(types)*/
+            {
+                let may_have_name_collisions = !context
+                    .flags
+                    .intersects(NodeBuilderFlags::UseFullyQualifiedType);
+                let mut result: Vec<Rc<Node>> = vec![];
+                let mut i: usize = 0;
+                for type_ in types {
+                    i += 1;
+                    let type_node =
+                        self.type_to_type_node_helper(type_checker, type_.clone(), context);
+                    result.push(type_node.into());
+                }
```

Comment:
Unless you're going to do something exciting with `i`, maybe you could iterate over the `.enumerate` of `types` instead of manually managing the index?